src/main/java/com/openshift/restclient/capability/resources/IPodExec.java

```diff
@@ -0,0 +1,150 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package com.openshift.restclient.capability.resources;
+
+import com.openshift.restclient.capability.ICapability;
+import com.openshift.restclient.capability.IStoppable;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Runs container exec
+ */
+public interface IPodExec extends ICapability{
+
+	/**
+	 * Execute a command on a named container in this pod
+	 * @param listener Listener for command output
+	 * @param options Options for the exec
+	 * @param commands A command to run and any arguments
+	 * @return A Handle to allow termination of the connection
+	 */
+	IStoppable start( IPodExecOutputListener listener, Options options, String... commands );
+
+
+	/**
+	 * A callback for exec output
+	 *
+	 */
+	interface IPodExecOutputListener{
+
+		/**
+		 * Callback received on initial connection
+		 */
+		void onOpen();
+
+		/**
+		 * Exec received stdout message
+		 * @param message
+		 */
+		void onStdOut(String message);
+
+		/**
+		 * Exec received stderr message
+		 * @param message
+		 */
+		void onStdErr(String message);
+
+		/**
+		 * Exec (channel 3) error message
+		 * @param message
+		 */
+		void onExecErr(String message);
```

Comment:
I can definitely get talk on that channel. Example of launching non-existent binary:
```
OPEN!
stdout: 
stderr: exec: "/bin/doesnotexist": stat /bin/doesnotexist: no such file or directory

execerr: error executing command in container: Error executing in Docker Container: -1
CLOSE!
```